embedded/drivers/include/can.h

```diff
@@ -0,0 +1,56 @@
+#ifndef _CAN_H__
+#define _CAN_H__
+
+#include <linux/can.h>
+#include <net/if.h>
+#include <semaphore.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h>
+
+#include <cstring>
+#include <thread>
+#include <vector>
```

Comment:
Pedantic, but let's try to keep includes that aren't required by the header file in the source file. Also drop the time.h include along with the timer variable

Reply:
Ok, dropped the timer var and time header file.

You're sure you don't want the headers to have all the includes so the source is cleaner and just says include "can.h"? I thought new members would be less intimidated and understand better having all defines and header files in the .h. If you notice rms.h is cleaner in my opinion too having the includes in can.h

What do you think @EUdds

Reply:
From [GCC](https://gcc.gnu.org/onlinedocs/cpp/Header-Files.html#:~:text=A%20header%20file%20is%20a,Header%20files%20serve%20two%20purposes)
"A header file is a file containing C declarations and macro definitions to be shared between several source files."

Headers are meant to be shared between source files. The includes that are specific to the source file might not need to be included in other sources that include the header file